plugins/modules/aci_rest.py

```diff
@@ -303,6 +309,28 @@ def update_qsl(url, params):
         return url + "?" + "&".join(["%s=%s" % (k, v) for k, v in params.items()])
 
 
+def add_annotation(annotation, payload):
+    """Add annotation to payload only if it has not already been added"""
+    if annotation:
+        for key, val in payload.items():
+            if key in ANNOTATION_UNSUPPORTED:
+                return
+            att = val.get("attributes", {})
+            if "annotation" not in att.keys():
+                att["annotation"] = annotation
+
+
+def add_annotation_xml(annotation, tree):
+    """Add annotation to payload xml only if it has not already been added"""
+    if annotation:
+        for element in tree.iter():
+            if element.tag in ANNOTATION_UNSUPPORTED:
+                return
+            ann = element.get("annotation")
+            if ann is None:
+                element.set("annotation", annotation)
```

Comment:
Should we write tests for this functionality?

Reply:
Good point, yeah I think we should.

Reply:
Done! and they pass 😸